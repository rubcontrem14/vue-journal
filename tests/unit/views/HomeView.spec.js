import { shallowMount } from "@vue/test-utils";
import HomeView from '@/views/HomeView'

describe("Pruebas en HomeView", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallowMount(HomeView);
    });

    test("debe de hacer match con el snapshot", () => {

        expect(wrapper.html()).toMatchSnapshot();

    });

    test("hacer click en un botÃ³n debe de redireccionar a no-entry", () => {

        const mockRouter = {
            push: jest.fn()
        }
        wrapper = shallowMount(HomeView, {
            global: {
                mocks:{
                    $router: mockRouter
                }
            }
        });
        wrapper.find('button').trigger('click')

        expect(mockRouter.push).toHaveBeenCalled()
        expect(mockRouter.push).toHaveBeenCalledWith({ name: "no-entry" })
    });
})