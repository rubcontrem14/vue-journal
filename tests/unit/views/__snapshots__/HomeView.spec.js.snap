// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en HomeView debe de hacer match con el snapshot 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <h1>Hola</h1><button class="btn btn-primary">Primary</button><button class="btn btn-secondary">Secondary</button><button class="btn btn-success">Success</button>
</div>
`;
